### PR TITLE
Add a new rule: HidesBottomBar

### PR DESCRIPTION
**Pull request by ykkd**
### About
- Add a new rule to check if HidesBottomBarWhenPushed is enabled

- HidesBottomBar rule is especially useful when the most of ViewControllers in your app need to be pushed with their bottom bar hidden.
  - Since the value of HidesBottomBarWhenPushed is `false` by default, I assume this rule can help prevent possible human errors.

- In order to implement the rule, I had to make change to IBDecodable, too.
  - I will fix `Packages.swift` and open this PR, once the one for [IBDecodable](https://github.com/IBDecodable/IBDecodable/pull/111)'s approved
  
  

### P.S.
- if appropriate, It'll help a lot if [suporting dynamic path](https://github.com/IBDecodable/IBLinter/pull/135) could be merged👀

Thanks for this wonderful tool!

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
OK, I've just released IBDecodable 0.5.0.

**Comment by ykkd** [comment]
@kateinoigakukun thanks again!

File: Sources/IBLinterKit/Rules/HidesBottomBarRule.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        func validate(xib: XibFile) -> [Violation] { return [] }
+
+        func validate(storyboard: StoryboardFile) -> [Violation] {
+            guard let scenes = storyboard.document.scenes else { return [] }
+            let viewControllers = scenes.compactMap { $0.viewController?.viewController }
+            return viewControllers.compactMap { validate(for: $0, file: storyboard) }
+        }
+
+        private func validate<T: InterfaceBuilderFile>(for viewController: ViewControllerProtocol, file: T) -> Violation? {
+            guard let customClass = viewController.customClass,
```
Comment (kateinoigakukun) [inline Sources/IBLinterKit/Rules/HidesBottomBarRule.swift:32]: It seems this rule is only applied for view controllers with custom classes, is this intentional? Should we warn a VC, which doesn't specify hidesBottomBarWhenPushed and doesn't have a custom class?

**Comment by ykkd** [comment]
@kateinoigakukun 
Thanks!
It was intentional but now that you ask me, I agree with you.
I suppose previous commits fixed the issue.

**Review by kateinoigakukun — APPROVED** [review]
Nice 😎

**Pull request merged**